src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -937,26 +940,63 @@
                                                      "code_id"
                                                      (current-code-id-fn (get request-options "environment")))))})))
 
+(defn parse-project-compile-data
+  "Parse data required to compile a catalog inside a project. Data required includes
+   * Root path to project
+   * modulepath
+   * hiera config
+   * project_name"
+  [request-options
+   bolt-projects-dir]
+  (let [versioned-project (:versioned_project request-options)
+        project-root (file-serving/get-project-root bolt-projects-dir versioned-project)
+        project-config (file-serving/read-bolt-project-config project-root)]
+    (-> request-options
+        (assoc "project_root" project-root)
+        (assoc "modulepath" (file-serving/get-project-modulepath project-config))
+        (assoc "hiera_config" (:hiera-config project-config))
+        (assoc "project_name" (:name project-config)))))
+
 (schema/defn ^:always-validate compile-fn :- IFn
   [jruby-service :- (schema/protocol jruby-protocol/JRubyPuppetService)
    current-code-id-fn :- IFn
-   boltlib-path :- (schema/maybe [schema/Str])]
+   boltlib-path :- (schema/maybe [schema/Str])
+   bolt-projects-dir :- (schema/maybe [schema/Str])]
```

Comment:
This must be a copy paste error? We dont expect this to be an array of paths right? Ultimately, the versioned_project we send along in the request would resolve to a single file path to a deployed bolt project?

Reply:
Oh, yep! good catch